bundles/org.openhab.binding.smartthings/contrib/smartthings/SmartApps/OpenHabAppV2.groovy

```diff
@@ -0,0 +1,1052 @@
+/**
+ *  OpenHabAppV2
+ *
+ *  Description
+ *   Provides two way communications between a Smartthings Hub and OpenHAB
+ *   Messages from OpenHAB with the following paths are supported and perform the following functions
+ *    /state  - returns the state of the specified device and attribute, i.e. on, off, 95
+ *    /update - Updates the state of the specified device and attribute
+ *    /discovery - Returns a list of the devices
+ *    /error - Returns error messages to OpenHAB for logging
+ *   Messages are sent to OpenHAB with the following paths
+ *    /smartthings/push - When an event occurs on a monitored device the new value is sent to OpenHAB
+ *
+ *  Authors
+ *   - rjraker@gmail.com - 1/30/17 - Modified for use with Smartthings
+ *   - st.john.johnson@gmail.com and jeremiah.wuenschel@gmail.com- original code for interface with another device
+ *
+ *  Copyright 2016 - 2020
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ */
+import groovy.json.JsonSlurper
+import groovy.json.JsonOutput
+import groovy.json.JsonBuilder
+import groovy.transform.Field
+
+// Massive lookup tree
+@Field CAPABILITY_MAP = [
+    "accelerationSensor": [
+        name: "Acceleration Sensor",
+        capability: "capability.accelerationSensor",
+        attributes: [
+            "acceleration"
+        ]
+    ],
```

Comment:
Is this list generated or do you have to update by hand? You should look into finding a way to generate it automatically if possible.

Reply:
This is based on the Smartthings capability and that is available as an xml file. In the future I can look at building an xslt that can transform it. It would also be nice to dynamically generate the **SUPPORTED_THING_TYPES_UIDS** set that is in SmartthingsBindingConstants.java. Are you aware of a way to do that?